crates/burn-cubecl/src/ops/qtensor.rs

```diff
@@ -230,19 +249,32 @@ where
             return execute_with_dtype!(tensor.dtype, E, into_data::<R, E>(tensor).await);
         }
 
-        let tensor = kernel::into_contiguous_aligned(tensor);
-        let mut data = match tensor.scheme() {
-            QuantScheme {
-                q_type: QuantInputType::QInt8,
-                ..
-            } => into_data::<R, i8>(tensor.clone()).await,
+        let (shape, dtype) = (tensor.shape.dims.to_vec(), tensor.dtype.clone());
```

Comment:
`dims.clone()` is more explicit/clear than `to_vec()` since it's already a vec. Also, dtype implements copy.

```suggestion
        let (shape, dtype) = (tensor.shape.dims.clone(), tensor.dtype);
```